DependencyInjection/LiipTranslationExtension.php

```diff
@@ -35,10 +36,22 @@ public function load(array $configs, ContainerBuilder $container)
         $config = $this->processConfiguration($configuration, $configs);
 
         $container->setParameter('liip_translation_config', $config);
-        $container->setParameter('liip.translation.persistence.class', $config['persistence']['class']);
-        $container->setParameter('liip.translation.persistence.options', $config['persistence']['options']);
 
         $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.yml');
+
+        $this->loadPersistenceConfiguration($config['persistence'], $container);
+    }
+
+    protected function loadPersistenceConfiguration(array $config, ContainerBuilder $container)
+    {
+        if (isset($config['service'])) {
+            $container->setAlias('liip.translation.persistence', $config['service']);
+        }
+
+        if (isset($config['class'])) {
+            $persistence = new Definition($config['class'], array($config['options']));
+            $container->setDefinition('liip.translation.persistence', $persistence);
+        }
```

Comment:
no, if you remove the definition if a service is configured, this happens before the parameters are checked.